modelci/finetuner/torch_data_module.py

```diff
@@ -3,120 +3,93 @@
-from typing import List, Optional, Tuple
-
-class TorchDataset(Dataset):
-    """Dataset with support of transforms for pytorch models.
-    """
-
-    def __init__(self, data: np.array, targets: List, transform: Optional[Compose] = None,
-                 target_transform: Optional[Compose] = None):
-        """
+import torchvision
+from torch.utils.data import random_split, DataLoader
+from torchvision.transforms import transforms
 
-        Args:
-            data (np.array): image data in numpy array format
-            targets (List): label data
-            transform (Optional[Compose]): data transform applied to image data
-            target_transform (Optional[Compose]):  data transform applied to label data
-        """
-        self.data = data
-        self.targets = targets
-        self.transform = transform
-        self.target_transform = target_transform
+# transforms
+from modelci.finetuner import OUTPUT_DIR
 
-    def __getitem__(self, index: int) -> Tuple[object, int]:
-        """
-        Reference: https://github.com/pytorch/vision/blob/master/torchvision/datasets/cifar.py#L105-L128
-        Args:
-            index (int): Index
-        Returns:
-            tuple: (Tuple[object, int]) where target is index of the target class.
-        """
-        img, target = self.data[index], self.targets[index]
+input_size = (224, 224)
 
-        # doing this so that it is consistent with all other datasets
-        # to return a PIL Image
-        img = Image.fromarray(img)
+train_transforms = transforms.Compose([
+    transforms.RandomResizedCrop(input_size),
+    transforms.RandomHorizontalFlip(),
+    transforms.ToTensor(),
+    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+])
 
-        if self.transform is not None:
-            img = self.transform(img)
-
-        if self.target_transform is not None:
-            target = self.target_transform(target)
-
-        return img, target
-
-    def __len__(self) -> int:
-        return len(self.data)
+test_transforms = transforms.Compose([
+    transforms.Resize(input_size),
+    transforms.CenterCrop(input_size),
+    transforms.ToTensor(),
+    transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225])
+])
 
 
+# noinspection PyAbstractClass
 class PyTorchDataModule(pl.LightningDataModule):
 
-    def __init__(self, dataset_name: str, input_size: Tuple[int, int]=(224,224), batch_size: int=8):
+    def __init__(self, dataset_name: str, batch_size: int = 8, num_workers=2, data_dir=OUTPUT_DIR):
         """
 
         Args:
-            dataset_name (str): name of dataset to be load, full lists of supported datasets: https://www.tensorflow.org/datasets/catalog/overview#all_datasets
-            input_size (Tuple[int, int]): input image size of target model
+            dataset_name (str): name of dataset to be load, full lists of supported datasets:
+                https://www.tensorflow.org/datasets/catalog/overview#all_datasets
```

Comment:
The supported datasets url should be https://pytorch.org/docs/stable/torchvision/datasets.html